src/Components/MemberLogged/SubscribeActivities/FormMemberSubscription/index.jsx

```diff
@@ -0,0 +1,129 @@
+import Button from 'Components/Shared/Button';
+import ConfirmModal from 'Components/Shared/ConfirmModal';
+import { getClasses } from 'Redux/Classes/thunks';
+import { addSubscriptions, getSubscriptions } from 'Redux/Subscriptions/thunks';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useHistory, useParams } from 'react-router-dom/cjs/react-router-dom.min';
+import Loader from 'Components/Shared/Loader';
+import { joiResolver } from '@hookform/resolvers/joi';
+import { useForm } from 'react-hook-form';
+import subscriptionSchema from 'Validations/subscription';
+import styles from './formMemberSubscription.module.css';
+import { resetState } from 'Redux/Subscriptions/actions';
+
+const FormMemberSubscription = () => {
+  const { id } = useParams();
+  const history = useHistory();
+  const { isPending } = useSelector((state) => state.classes);
+  const member = useSelector((state) => state.loginMembers.data);
+  const { success } = useSelector((state) => state.subscriptions);
+  const dispatch = useDispatch();
+
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [classes, setClasses] = useState([]);
+  const {
+    handleSubmit,
+    register,
+    formState: { errors }
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(subscriptionSchema),
+    defaultValues: {
+      members: member._id,
+      classes: ''
+    }
+  });
+
+  useEffect(() => {
+    getSubscriptions(dispatch);
+    getClasses(dispatch).then((data) => {
+      filterClassAndOrder(data);
+    });
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (success) {
+      history.push('/user/members/subscribe-class');
+      dispatch(resetState());
+    }
+  }, [success]);
+
+  const filterClassAndOrder = (data) => {
+    const filteredClasses = data.filter(
+      (item) => !item.deleted && item.activity && item.activity._id === id
+    );
+    filteredClasses.sort((a, b) => {
+      const dayComparison = a.day.localeCompare(b.day);
+      if (dayComparison !== 0) {
+        return dayComparison;
+      }
+      return a.time.localeCompare(b.time);
+    });
+    setClasses(filteredClasses);
+  };
+
+  const handleShowConfirmModal = async () => {
+    if (!showConfirmModal) {
+      setShowConfirmModal(true);
+    } else {
+      setShowConfirmModal(false);
+    }
+  };
```

Comment:
Here (the function above) there is no need to do an async function. Also you can improve the function by setting the oposite state like this:
```suggestion
  const handleShowConfirmModal = () => {
    setShowConfirmModal(!showConfirmModal);
  };
```